source/time_stepping/interface.cc

```diff
@@ -26,137 +26,73 @@ namespace aspect
-                          this->get_solution());
-
-                this->get_material_model().evaluate(in, out);
-
-                if (this->get_parameters().formulation_temperature_equation
-                    == Parameters<dim>::Formulation::TemperatureEquation::reference_density_profile)
-                  {
-                    // Overwrite the density by the reference density coming from the
-                    // adiabatic conditions as required by the formulation
-                    for (unsigned int q=0; q<n_q_points; ++q)
-                      out.densities[q] = this->get_adiabatic_conditions().density(in.position[q]);
-                  }
-                else if (this->get_parameters().formulation_temperature_equation
-                         == Parameters<dim>::Formulation::TemperatureEquation::real_density)
-                  {
-                    // use real density
-                  }
-                else
-                  AssertThrow(false, ExcNotImplemented());
-
-
-                // Evaluate thermal diffusivity at each quadrature point and
-                // calculate the corresponding conduction timestep, if applicable
-                for (unsigned int q=0; q<n_q_points; ++q)
-                  {
-                    const double k = out.thermal_conductivities[q];
-                    const double rho = out.densities[q];
-                    const double c_p = out.specific_heat[q];
-
-                    Assert(rho * c_p > 0,
-                           ExcMessage ("The product of density and c_P needs to be a "
-                                       "non-negative quantity."));
-
-                    const double thermal_diffusivity = k/(rho*c_p);
-
-                    if (thermal_diffusivity > 0)
-                      {
-                        min_local_conduction_timestep = std::min(min_local_conduction_timestep,
-                                                                 this->get_parameters().CFL_number*pow(cell->minimum_vertex_distance(),2.)
-                                                                 / thermal_diffusivity);
-                      }
-                  }
-              }
-          }
 
-      const double max_global_speed_over_meshsize
-        = Utilities::MPI::max (max_local_speed_over_meshsize, this->get_mpi_communicator());
+    template <int dim>
+    void
+    Interface<dim>::
+    parse_parameters (ParameterHandler &/*prm*/)
+    {
+    }
 
-      double min_convection_timestep = std::numeric_limits<double>::max();
-      double min_conduction_timestep = std::numeric_limits<double>::max();
 
-      if (max_global_speed_over_meshsize != 0.0)
-        min_convection_timestep = this->get_parameters().CFL_number / (this->get_parameters().temperature_degree * max_global_speed_over_meshsize);
 
-      if (this->get_parameters().use_conduction_timestep)
-        min_conduction_timestep = - Utilities::MPI::max (-min_local_conduction_timestep, this->get_mpi_communicator());
```

Comment:
You cancelled out the sign, which honestly make a lot more sense to me, but I am wondering why a minus sign was used here in the first place.

Reply:
We computed ``x = min(x[i])`` as ``x = - max(-x[i])`` because min() did not exist (but that was a long time ago).

Reply:
ah, wow I never though a max function would exists without a min function :s